File: src/main/java/com/causr/dto/AnomalyDto.java
package com.causr.dto;

public class AnomalyDto {

    private String service;

    private String type;

    private String severity;

    private String message;

    private long timestamp;

	public String getService() {
		return service;
	}

	public void setService(String service) {
		this.service = service;
	}

	public String getType() {
		return type;
	}

	public void setType(String type) {
		this.type = type;
	}

	public String getSeverity() {
		return severity;
	}

	public void setSeverity(String severity) {
		this.severity = severity;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public long getTimestamp() {
		return timestamp;
	}

	public void setTimestamp(long timestamp) {
		this.timestamp = timestamp;
	}
	
    
}

File: causr-ui/src/lib/anomaly-stream/anomalyDto.ts
/** Matches {@code AnomalyDto} JSON from Causr-log-processor {@code /topic/anomalies}. */
export type AnomalyDto = {
  service: string
  type: string
  severity: string
  message: string
  timestamp: number
}

export function parseAnomalyDto(raw: string): AnomalyDto | null {
  try {
    const v = JSON.parse(raw) as unknown
    if (!v || typeof v !== 'object') return null
    const o = v as Record<string, unknown>
    const service = typeof o.service === 'string' ? o.service : 'unknown'
    const type = typeof o.type === 'string' ? o.type : 'UNKNOWN'
    const severity = typeof o.severity === 'string' ? o.severity : 'warning'
    const message = typeof o.message === 'string' ? o.message : ''
    const ts = o.timestamp
    const timestamp = typeof ts === 'number' && Number.isFinite(ts) ? ts : Date.now()
    return { service, type, severity, message, timestamp }
  } catch {
    return null
  }
}

export function anomalyId(dto: AnomalyDto): string {
  return `${dto.timestamp}-${dto.service}-${dto.type}`
}
